### PR TITLE
PoS documentation

### PR DESCRIPTION
**Pull request by sanchaymittal**

### CODE REVIEW

File: docs/develop/ethereum-matic/pos/contract-interfaces.md [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+  - matic
+image: https://matic.network/banners/matic-network-16x9.png 
+---
+
+### IRootChainManager
+
+```jsx
+pragma solidity "0.6.6";
+
+interface IRootChainManager {
```
Comment (arthcp) [inline docs/develop/ethereum-matic/pos/contract-interfaces.md:17]: We can update to latest interface-
https://github.com/maticnetwork/pos-portal/blob/v1.0.0/contracts/root/RootChainManager/IRootChainManager.sol

File: docs/develop/ethereum-matic/pos/erc1155-desposit-withdraw.md [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+  - docs
+  - matic
+image: https://matic.network/banners/matic-network-16x9.png 
+---
+## High Level Flow
+
+Deposit ERC1155 -
+
+1. ***Approve*** ***RootChainManager*** contract to spend the tokens that have to be deposited.
+2. Make ***deposit*** or ***depositFor*** call on ***RootChainManager***.
```
Comment (arthcp) [inline docs/develop/ethereum-matic/pos/erc1155-desposit-withdraw.md:15]: there is only `depositFor` method deposit has been removed, applies for all token types

File: docs/develop/ethereum-matic/pos/erc1155-desposit-withdraw.md [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+1. ***Burn*** tokens on matic chain.
+2. Call ***exit*** function on ***RootChainManager*** to submit proof of burn transaction. This call can be made ***after checkpoint*** is submitted for the block containing burn transaction.
+
+## Step Details
+
+---
+
+### Configuring Matic SDK
+
+Install Matic SDK (***2.0.0-beta.12)***
```
Comment (arthcp) [inline docs/develop/ethereum-matic/pos/erc1155-desposit-withdraw.md:28]: change matic js version to 2.0.2

Comment (arthcp) [reply docs/develop/ethereum-matic/pos/erc1155-desposit-withdraw.md:28]: This applies to all token types

File: docs/develop/ethereum-matic/pos/erc1155-desposit-withdraw.md [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+
+```jsx
+const MaticPOSClient = require('@maticnetwork/maticjs').MaticPOSClient
+
+const maticPOSClient = new MaticPOSClient({
+  maticProvider: config.MATIC_PROVIDER,
+  parentProvider: config.PARENT_PROVIDER,
+  rootChain: config.PLASMA_ROOTCHAIN_ADDRESS,
+  posRootChainManager: config.POS_ROOT_CHAIN_MANAGER_ADDRESS,
+})
```
Comment (arthcp) [inline docs/develop/ethereum-matic/pos/erc1155-desposit-withdraw.md:50]: Should be updated according to latest examples. Multiple changes like `rootChain` is not needed, `network` and `version` were added etc.
https://github.com/maticnetwork/matic.js/tree/v2.0.2/examples/POS-client
This applies to all token types

File: docs/develop/ethereum-matic/pos/support-multiple-token.md [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+
+# Specification
+
+### `ITokenPredicate.sol`
+
+```jsx
+abstract contract ITokenPredicate {
+    function lockTokens(address user, address rootToken, bytes memory tokenData) public virtual;
+
+    function validateBurn(bytes memory burnLogRLP) public pure virtual;
```
Comment (arthcp) [inline docs/develop/ethereum-matic/pos/support-multiple-token.md:32]: `validateBurn` is removed, `exitTokens` function performs all these checks

File: docs/develop/ethereum-matic/pos/getting-started.md [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+### Child Chain contracts
+
+- ***rootToChild*** and ***childToRoot*** token mapping should be stored.
+- ***StateReceiver*** should be able to call ***onStateReceive*** function.
+- ***ChildToken*** should have ***deposit*** and ***withdraw*** methods.
+
+### General state
+
+- We will allow exits of `stateRoot` which will be logged into `burn` tx and destroyed on Child contracts and while exiting users will provide proof for `stateRoot` and also provide the `stateData` and on the RootChain contract mapped contract will handle the state write mech similar to `Predicate` Which holds true for state deposits as well.
+- Will write a sample state deposit/exit contract and doc for the same later once we've completed the simple `erc20/erc721.`
```
Comment (arthcp) [inline docs/develop/ethereum-matic/pos/getting-started.md:47]: General state part can be removed. This is internal plan, not implemented.

File: docs/develop/ethereum-matic/pos/contract-interfaces.md (hunk 1 above)
Comment (sanchaymittal) [reply docs/develop/ethereum-matic/pos/contract-interfaces.md:17]: Done

File: docs/develop/ethereum-matic/pos/erc1155-desposit-withdraw.md (hunk 2 above)
Comment (sanchaymittal) [reply docs/develop/ethereum-matic/pos/erc1155-desposit-withdraw.md:15]: got it.

**Pull request merged**